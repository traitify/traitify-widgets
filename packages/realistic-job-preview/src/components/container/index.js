import PropTypes from "prop-types";
import Context from "lib/context";
import ErrorBoundary from "./error-boundary";
import Theme from "./theme";
import useReducer from "./use-reducer";

function Container({children, className = null, ...props}) {
  const state = useReducer({...props});

  return (
    <Context.Provider value={state}>
      <Theme className={className}>
        <ErrorBoundary>{children}</ErrorBoundary>
      </Theme>
    </Context.Provider>
  );
}

Container.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string
};

export default Container;
