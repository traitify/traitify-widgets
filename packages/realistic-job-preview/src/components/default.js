import useAssessment from "lib/hooks/use-assessment";
import Results from "./results";
import Survey from "./survey";

export default function Default() {
  const assessment = useAssessment();

  if(!assessment) { return null; }
  if(assessment.completedAt) { return <Results />; }

  return <Survey />;
}
