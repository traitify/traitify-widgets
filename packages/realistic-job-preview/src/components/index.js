import Container from "./container";
import Default from "./default";

export default {
  Container,
  Default
};
