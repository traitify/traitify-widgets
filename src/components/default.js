import useActive from "lib/hooks/use-active";
import Results from "./results";
import Survey from "./survey";

export default function Default() {
  const active = useActive();

  if(!active) { return null; }
  if(active.loading) { return null; }
  if(active.completed) { return <Results />; }

  return <Survey />;
}
