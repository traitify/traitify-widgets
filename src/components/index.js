import Container from "./container";
import Default from "./default";
import PersonalityArchetypeHeading from "./personality/archetype/heading";
import PersonalityArchetypeSkills from "./personality/archetype/skills";
import PersonalityArchetypeTips from "./personality/archetype/tips";
import PersonalityBaseDetails from "./personality/base/details";
import PersonalityBaseHeading from "./personality/base/heading";
import PersonalityCareerContainer from "./personality/career/container";
import PersonalityCareerDetails from "./personality/career/details";
import PersonalityCareerFilter from "./personality/career/filter";
import PersonalityCareerList from "./personality/career/list";
import PersonalityCareerModal from "./personality/career/modal";
import PersonalityDimensionChart from "./personality/dimension/chart";
import PersonalityDimensionDetails from "./personality/dimension/details";
import PersonalityDimensionList from "./personality/dimension/list";
import PersonalityRecommendationChart from "./personality/recommendation/chart";
import PersonalityTraitDetails from "./personality/trait/details";
import PersonalityTraitList from "./personality/trait/list";
import PersonalityTypeChart from "./personality/type/chart";
import PersonalityTypeList from "./personality/type/list";
import AttractReport from "./report/attract";
import CandidateReport from "./report/candidate";
import EmployeeReport from "./report/employee";
import ManagerReport from "./report/manager";

export default {
  Container,
  Default,
  Personality: {
    Archetype: {
      Heading: PersonalityArchetypeHeading,
      Skills: PersonalityArchetypeSkills,
      Tips: PersonalityArchetypeTips
    },
    Base: {
      Details: PersonalityBaseDetails,
      Heading: PersonalityBaseHeading
    },
    Career: {
      Container: PersonalityCareerContainer,
      Details: PersonalityCareerDetails,
      Filter: PersonalityCareerFilter,
      List: PersonalityCareerList,
      Modal: PersonalityCareerModal
    },
    Dimension: {
      Chart: PersonalityDimensionChart,
      Details: PersonalityDimensionDetails,
      List: PersonalityDimensionList
    },
    Recommendation: {
      Chart: PersonalityRecommendationChart
    },
    Trait: {
      Details: PersonalityTraitDetails,
      List: PersonalityTraitList
    },
    Type: {
      Chart: PersonalityTypeChart,
      List: PersonalityTypeList
    }
  },
  Report: {
    Attract: AttractReport,
    Candidate: CandidateReport,
    Employee: EmployeeReport,
    Manager: ManagerReport
  }
};
