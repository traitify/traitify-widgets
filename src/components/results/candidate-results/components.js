export Guide from "components/results/guide";
export PersonalityArchetype from "components/results/personality/archetype/heading";
export PersonalityTips from "components/results/personality/archetype/tips";
export PersonalityDimensionChart from "components/results/personality/dimension/chart";
export PersonalityDimensionColumn from "components/results/personality/dimension/column";
export PersonalityDimensionColumns from "components/results/personality/dimension/chart";
export PersonalityDimensionDetails from "components/results/personality/dimension/details";
export PersonalityDimensions from "components/results/personality/dimension/list";
export PersonalityRecommendationChart from "components/results/personality/recommendation/chart";
export PersonalityTraits from "components/results/personality/trait/list";
export CandidateResults from "./index";
