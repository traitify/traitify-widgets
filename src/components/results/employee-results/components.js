export PersonalityArchetype from "components/results/personality/archetype/heading";
export PersonalitySkills from "components/results/personality/archetype/skills";
export PersonalityTips from "components/results/personality/archetype/tips";
export PersonalityDimensionChart from "components/results/personality/dimension/chart";
export PersonalityDimensionColumn from "components/results/personality/dimension/column";
export PersonalityDimensionColumns from "components/results/personality/dimension/chart";
export PersonalityDimensionDetails from "components/results/personality/dimension/details";
export PersonalityDimensions from "components/results/personality/dimension/list";
export EmployeeResults from "./index";
