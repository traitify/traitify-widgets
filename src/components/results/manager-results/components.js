export Guide from "components/results/guide";
export PersonalityRecommendationChart from "components/results/personality/recommendation/chart";
export ManagerResults from "./index";
