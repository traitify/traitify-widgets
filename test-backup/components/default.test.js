import {Component} from "components/default";
import ComponentHandler from "support/component-handler";

jest.mock("components/results", () => (() => (<div className="mock">Results</div>)));
jest.mock("components/slide-deck", () => (() => (<div className="mock">Slide Deck</div>)));
jest.mock("lib/with-traitify");

describe("Default", () => {
  it("renders results", () => {
    const component = new ComponentHandler(<Component isReady={(type) => (type === "results")} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders slide deck", () => {
    const component = new ComponentHandler(<Component isReady={(type) => (type === "slides")} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders div if not ready", () => {
    const component = new ComponentHandler(<Component isReady={() => (false)} />);

    expect(component.tree).toMatchSnapshot();
  });
});
