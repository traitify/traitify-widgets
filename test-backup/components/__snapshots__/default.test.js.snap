// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default renders div if not ready 1`] = `<div />`;

exports[`Default renders results 1`] = `
<div
  className="mock"
>
  Results
</div>
`;

exports[`Default renders slide deck 1`] = `
<div
  className="mock"
>
  Slide Deck
</div>
`;
